bpf/lib/nodeport.h

```diff
@@ -2544,15 +2567,29 @@ int tail_nodeport_nat_ingress_ipv4(struct __ctx_buff *ctx)
 	/* If we're not in full DSR mode, reply traffic from remote backends
 	 * might pass back through the LB node and requires revDNAT.
 	 *
-	 * Also let rev_nodeport_lb4() redirect EgressGW reply traffic into
-	 * tunnel (see there for details).
+	 * Also let nodeport_rev_dnat_ingress_ipv4() redirect EgressGW
+	 * reply traffic into tunnel (see there for details).
 	 */
-	ep_tail_call(ctx, CILIUM_CALL_IPV4_NODEPORT_REVNAT);
-#else
-	/* There's no reason to continue in the RevDNAT path, just recircle back. */
+	ret = invoke_traced_tailcall_if(__not(is_defined(HAVE_LARGE_INSN_LIMIT)),
+					CILIUM_CALL_IPV4_NODEPORT_REVNAT,
+					nodeport_rev_dnat_ingress_ipv4,
+					&trace, &ext_err);
+		if (IS_ERR(ret))
+			goto drop_err;
+
+		/* No redirect needed: */
+		if (ret == CTX_ACT_OK)
+			goto recircle;
+
+		/* Redirected to egress interface: */
+		edt_set_aggregate(ctx, 0);
```

Comment:
previously we set it only when !LXC. Can this be called from LXC?

Reply:
I added the following to the patch description, thank you!

>    Note that CILIUM_CALL_IPV*_NODEPORT_NAT_INGRESS is never invoked from  bpf_lxc, so we can remove the #ifdef's around edt_set_aggregate().